ingest/rules/transform.smk

```diff
@@ -0,0 +1,97 @@
+"""
+This part of the workflow handles transforming the data into standardized
+formats and expects input file
+
+    sequences_ndjson = "data/sequences.ndjson"
+
+This will produce output files as
+
+    metadata = "results/metadata.tsv"
+    sequences = "results/sequences.fasta"
+
+Parameters are expected to be defined in `config.transform`.
+"""
+
+
+rule fetch_general_geolocation_rules:
+    output:
+        general_geolocation_rules="data/general-geolocation-rules.tsv",
+    params:
+        geolocation_rules_url=config["transform"]["geolocation_rules_url"],
+    shell:
+        """
+        curl {params.geolocation_rules_url} > {output.general_geolocation_rules}
+        """
+
+
+rule concat_geolocation_rules:
+    input:
+        general_geolocation_rules="data/general-geolocation-rules.tsv",
+        local_geolocation_rules=config["transform"]["local_geolocation_rules"],
+    output:
+        all_geolocation_rules="data/all-geolocation-rules.tsv",
+    shell:
+        """
+        cat {input.general_geolocation_rules} {input.local_geolocation_rules} >> {output.all_geolocation_rules}
+        """
+
+
+rule transform:
+    input:
+        sequences_ndjson="data/sequences.ndjson",
+        all_geolocation_rules="data/all-geolocation-rules.tsv",
+        annotations=config["transform"]["annotations"],
+    output:
+        metadata="results/metadata.tsv",
+        sequences="results/sequences.fasta",
+    log:
+        "logs/transform.txt",
+    params:
+        field_map=config["transform"]["field_map"],
```

Comment:
Ah, just noticed that this is handled a little differently in the [pathogen-repo-template](https://github.com/nextstrain/pathogen-repo-template/blob/b8ae886b25877a218ad50380fb44f8825d50aedb/ingest/rules/curate.smk#L71) to simplify the [field_map in the config YAML](https://github.com/nextstrain/pathogen-repo-template/blob/b8ae886b25877a218ad50380fb44f8825d50aedb/ingest/config/defaults.yaml#L47-L63)